### ISSUE TITLE
Which lighting profile did the original game use?

### ISSUE THREAD

**Issue by Dalai-** [fredlllll/ArxLibertatisLightingCalculator#1]
Or is the lighting logic from THEO not really known 100%?

**Comment by fredlllll** [comment]
Danae is the lighting profile that i wrote after the code that was released with the arx source code. i have not tested this, but it should produce the same results as the original game has, or at least close it. theo actually didnt precalculate the lighting i think. that was done in danae.

**Comment by Dalai-** [comment]
Ah okay, thanks. I noticed the [LightingCalculatorDanae.cs](https://github.com/fredlllll/ArxLibertatisLightingCalculator/blob/master/ArxLibertatisLightingCalculator/LightingCalculatorDanae.cs) doesn't have any shadow or polygon intersection tests like [LightingCalculatorDistanceAngleShadowBase.cs](https://github.com/fredlllll/ArxLibertatisLightingCalculator/blob/master/ArxLibertatisLightingCalculator/LightingCalculatorDistanceAngleShadowBase.cs) has. So that means in the original game all the precalculated lighting would just go through walls and not have shadows? Or bounced light? They must have been careful about light placement if that's the case... I'm surprised the end result looks as good as it does!

**Comment by fredlllll** [comment]
yes, in certain areas it is also visible that light shines through walls. notice that the torch in greus cave is off. this is becuase it would shine through to the prison cell the player spawns in. and there are other instances where lights were carefully placed to not influence things through walls

**Comment by Dalai-** [comment]
Hmm, it seems the "Ray-Launching" option in Danae does do collision tests for lights... screenshot of the option here: https://wiki.arx-libertatis.org/images/0/00/Danae_Options.jpg

**Comment by fredlllll** [comment]
i never looked at the UI of danae. just the code i found buried in the codebase. the option would suggest that it was using collision detection, but from what i see in the game, it wasnt used. also didnt find anything about it in the light calculation code

**Comment by Dalai-** [comment]
Looking at the DANAE code, that "Ray-Launching" tickbox causes the snippet below to run in ARX_EERIE_LIGHT_Make (part of EERIEPrecalcLights logic).

```
			//---------------------- start MODE_RAYLAUNCH
			if ((ModeLight & MODE_RAYLAUNCH) && !(light->extras & EXTRAS_NOCASTED))
			{
				EERIE_3D orgn, dest, hit;
				orgn.x = light->pos.x;
				orgn.y = light->pos.y;
				orgn.z = light->pos.z;
				dest.x = ep->v[i].sx;
				dest.y = ep->v[i].sy;
				dest.z = ep->v[i].sz;

				if (ModeLight & MODE_SMOOTH)
					fRes *= my_CheckInPoly(ep->v[i].sx, ep->v[i].sy, ep->v[i].sz, ep, light);
				else
					fRes *= Visible(&orgn, &dest, ep, &hit);
			}
```

**Comment by fredlllll** [comment]
its been a while since i wrote this, so i dont remember seeing this. but i can for sure say it wasnt used for the final lighting, as using my shadow casting lighting adds shadows to so many things. but because its vertex lighting it looks weird at times, so that could be a reason why they didnt use it
